test/tests/src/send.cpp

```diff
@@ -0,0 +1,124 @@
+#include <cassert>
+#include <chrono>
+#include <iostream>
+#include <random>
+#include "pack_unpack.hpp"
+#include "test.hpp"
+
+using namespace CANlib;
+using namespace map1;
+using namespace map2;
+using namespace J;
+using namespace CC;
+using namespace std::chrono_literals;
+
+extern CAN can1;
+extern CAN can2;
+extern CAN can3;
+
+#define CREATE_TEST(ID, can, BITMASK)                                         \
+  static void testSend##ID() {                                                \
+    ID##_T ID##_input;                                                        \
+    Frame f0;                                                                 \
+    for (int i = 0; i < 8; ++i) {                                             \
+      f0.data[i] = distribution(generator);                                   \
+    }                                                                         \
+    uint64_t bitstring0;                                                      \
+    to_bitstring((uint8_t *)f0.data, &bitstring0);                            \
+    bitstring0 &= BITMASK;                                                    \
+    ID##_input.pack(f0);                                                      \
+    can.clear();                                                              \
+    if (ID##_input.get_period() != 0ms) {                                     \
+      auto starting_time = Clock::now();                                      \
+      auto tmp           = starting_time;                                     \
+      while (Clock::now() - starting_time <= ID##_input.get_period() + 5ms) { \
+        send(tmp, &ID##_input);                                               \
```

Comment:
isn't checking the period redundant?

Reply:
Yes, the point of this is to not over-send, at the end we also check can.framesReceived() == 1.